common/djangoapps/config_models/models.py

```diff
@@ -22,46 +69,108 @@ class ConfigurationModel(models.Model):
 
     class Meta(object):  # pylint: disable=missing-docstring
         abstract = True
+        ordering = ("-change_date", )
+
+    objects = ConfigurationModelManager()
+
+    KEY_FIELDS = ()
 
     # The number of seconds
     cache_timeout = 600
 
-    change_date = models.DateTimeField(auto_now_add=True)
-    changed_by = models.ForeignKey(User, editable=False, null=True, on_delete=models.PROTECT)
-    enabled = models.BooleanField(default=False)
+    change_date = models.DateTimeField(auto_now_add=True, verbose_name=_("Change date"))
+    changed_by = models.ForeignKey(
+        User,
+        editable=False,
+        null=True,
+        on_delete=models.PROTECT,
+        # Translators: this label indicates the name of the user who made this change:
+        verbose_name=_("Changed by"),
+    )
+    enabled = models.BooleanField(default=False, verbose_name=_("Enabled"))
 
     def save(self, *args, **kwargs):
         """
         Clear the cached value when saving a new configuration entry
         """
         super(ConfigurationModel, self).save(*args, **kwargs)
-        cache.delete(self.cache_key_name())
+        cache.delete(self.cache_key_name(*[getattr(self, key) for key in self.KEY_FIELDS]))
+        if self.KEY_FIELDS:
+            cache.delete(self.key_values_cache_key_name())
 
     @classmethod
-    def cache_key_name(cls):
+    def cache_key_name(cls, *args):
         """Return the name of the key to use to cache the current configuration"""
-        return 'configuration/{}/current'.format(cls.__name__)
+        if cls.KEY_FIELDS != ():
+            if len(args) != len(cls.KEY_FIELDS):
+                raise TypeError(
+                    "cache_key_name() takes exactly {} arguments ({} given)".format(len(cls.KEY_FIELDS), len(args))
+                )
+            return u'configuration/{}/current/{}'.format(cls.__name__, u','.join(args))
```

Comment:
I appear to have led you astray here:

    >>> u','.join([1, 'foo', object()])
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
    TypeError: sequence item 0: expected string or Unicode, int found